identity/src/commonMain/kotlin/com/android/identity/util/ByteArrayUtil.kt

```diff
@@ -0,0 +1,234 @@
+package com.android.identity.util
+
+/**
+ * ByteArray extension functions for consistent reading and writing commonly used types.
+ */
+
+fun ByteArray.putInt8(offset: Int, value: Int, validRange: IntRange = Byte.MIN_VALUE..Byte.MAX_VALUE) {
```

Comment:
What's up with having `validRange` as a parameter? When would someone ever not use the default?

Reply:
This was one of my suggestions. There are places in the code where the valid range is much more limited. Being able to specify the valid range here would save an extra check and make things cleaner.